### ISSUE TITLE
`kill ENOSYS` error on Windows

### ISSUE THREAD

**Issue by BrooklynKing** [fabiospampinato/when-exit#1]
Same issue as [this one](https://github.com/tapjs/signal-exit/issues/54) in `signal-exits`. Af a fix they usings `SIGINT` instead of `SIGHUP` for Windows (https://github.com/tapjs/signal-exit/blob/main/src/index.ts#L149). From what I see, easiest way would be to introduce similar override of signal [here](https://github.com/fabiospampinato/when-exit/blob/master/src/node/interceptor.ts#L41)

**Comment by fabiospampinato** [comment]
@BrooklynKing Can you explain how this issue happens exactly?

I see from the code that a `SIGHUP` signal should be reemitted only if Node itself tells us that a it got a `SIGHUP` signal:

https://github.com/fabiospampinato/when-exit/blob/c70a8a1ada1c688accb1d58ab77ac08de1af892e/src/node/interceptor.ts/#L53

So the only way to make this happen, as I'm understanding it, is that somebody else gives Node a `SIGHUP` signal, Node tells us that it got a `SIGHUP` signal, and finally we tell Node to exit with a `SIGHUP` signal again, after we've called our callbacks.

Is that correct? If so isn't the actual problem here whoever is giving Node a `SIGHUP` at the start?

**Comment by BrooklynKing** [comment]
@fabiospampinato thx for answering!

In our case we got this error when user clicked on “x” of the Windows terminal.
From what I see in `signal-exit` repo, they have just checks when they need to pass `SIGHUP` signal. If it's `Win32` then they use `SIGINT` instead. 

Here is example their codebase - https://github.com/tapjs/signal-exit/blob/03dd77a96caa309c6a02c59274d58c812a2dce45/index.js#L127

In current version they changed it a bit https://github.com/tapjs/signal-exit/blob/main/src/index.ts#L149 + https://github.com/tapjs/signal-exit/blob/main/src/index.ts#L193

**Comment by fabiospampinato** [comment]
Thanks for the explanation.

I'm a bit conflicted on this, because I don't think this is the right level of abstraction where to put that code, unless I'm missing something.

Like I think one property that this library should have is that if you load it, and don't actually register any callbacks, then it doesn't change anything about your program. Don't you agree with this? If you don't for what reason?

Now if your program is exiting because of a SIGHUP, and a SIGHUP causes Node to throw for whatever weird reason, why should loading this library and registering no callbacks with it change anything?

It feels like this problem should be addressed probably, I've no idea if in Node or in Windows Terminal or in Windows itself or wherever else, but while a bit annoying fixing this in `when-exit` doesn't seem right. Probably you and I should have a `import 'normalize-sighup'` at the entrypoint of our CLIs or something.

**Comment by BrooklynKing** [comment]
I agree that it would be better if we could have something like `import 'normalize-sighup'`. But at the end of the day, it's your codebase that triggers `process.exit` with `SIGHUP`. If you don't want to handle this case - fair enough.

**Comment by fabiospampinato** [comment]
> But at the end of the day, it's your codebase that triggers process.exit with SIGHUP

Where am I explicitly triggering a `SIGHUP`? This library is transparent to the signals that get triggered, if you don't use you'll still see the `SIGHUP`, no?

To be clear if you are getting this problem in any of my CLIs I'll address this issue in all of them, since it seems a reasonable problem worth addressing on my end.

**Comment by BrooklynKing** [comment]
https://github.com/fabiospampinato/when-exit/blob/master/src/node/interceptor.ts#L53, you getting `SIGHUP` and then you triggering exit here (https://github.com/fabiospampinato/when-exit/blob/master/src/node/interceptor.ts#L39)

We using https://github.com/fabiospampinato/tiny-updater

**Comment by fabiospampinato** [comment]
You understand that if one just didn't import `tiny-updater` one would have the same problem right?

**Comment by BrooklynKing** [comment]
Yes, i understand that. 

Maybe you can summarize your position about this issue, so I can decide on my next steps?

**Comment by BrooklynKing** [comment]
We solved our issue in a different manner, sorry for bothering.

**Comment by fabiospampinato** [comment]
I've changed my mind on this, on the speculation that it's possible to load `when-exit`, don't do anything with it, and seeing this problem if something external to Node somehow gives it a `SIGHUP` signal, even though without loading `when-exit` one wouldn't see this problem.

I wasn't actually able to reproduce the issue by closing Windows Terminal on my end, so I'm not sure if that scenario is actually possible. Presumably it is, given what you are saying.

Basically this edge case should be handled in v2.1.2. 

Thanks.
